File: src/screen.py
from abc import abstractmethod, ABC
import threading

class Screen(threading.Thread, ABC):

    @abstractmethod
    def get_data(self):
        return self.data

    @abstractmethod
    def run(self):
        pass

def create(debug):
    from screendbg import ScreenDebug
    return ScreenDebug()

File: src/screen/screen.cc
#include "screen/screen.hh"

#include <memory>
#include <thread>
using namespace std;

#include "globals.hh"
#include "screen/screen_debug.hh"

std::unique_ptr<Screen>
Screen::create() { 
#ifdef DEBUG
    return std::make_unique<ScreenDebug>();
#else
    return std::make_unique<ScreenRPI>();
#endif
}

void
Screen::run()
{
    for (;;) {
        screen_queue.execute(*this);
        this_thread::sleep_for(10ms);
    }
}
